javers-core/src/test/groovy/org/javers/guava/multiset/MultisetChangeAppenderTest.groovy

```diff
@@ -0,0 +1,118 @@
+        def rightNode = buildGraph(new SnapshotEntity(multiSetOfPrimitives: HashMultiset.create(rightList)))
+
+        def change = multisetChangeAppender().calculateChanges(
+                new RealNodePair(leftNode, rightNode), getProperty(SnapshotEntity, "multiSetOfPrimitives"))
+
+        then:
+        change == null
+
+        where:
+        leftList   | rightList
+        []         | []
+        ["1", "2"] | ["1", "2"]
+        ["1", "2"] | ["2", "1"]
+    }
+
+    def "should append ValueAdded in Multiset of Values"() {
+        given:
+        def leftNode = buildGraph(new SnapshotEntity(multiSetValueObject: HashMultiset.create(leftList)))
+        def rightNode = buildGraph(new SnapshotEntity(multiSetValueObject: HashMultiset.create(rightList)))
+
+        when:
+        def change = multisetChangeAppender()
+                .calculateChanges(new RealNodePair(leftNode, rightNode), getProperty(SnapshotEntity, "multiSetValueObject"))
+
+        then:
+        assertThat(change)
+                .hasSize(1)
+                .hasValueAdded(expectedVal)
+        where:
+        leftList <<  [[new DummyAddress(city: "New York"),
+                       new DummyAddress(city: "New York")] ]
+        rightList << [[new DummyAddress(city: "New York"),
+                       new DummyAddress(city: "New York"),
+                       new DummyAddress(city: "Buffalo")] ]
+        expectedVal << [valueObjectId(1, SnapshotEntity, "multiSetValueObject/4057abef011cfffbdbaa632f49dcef56")]
+    }
+
+    def "should append ValueRemoved in Multiset of Values"() {
+        given:
+        def leftNode = buildGraph(new SnapshotEntity(multiSetValueObject: HashMultiset.create(leftList)))
+        def rightNode = buildGraph(new SnapshotEntity(multiSetValueObject: HashMultiset.create(rightList)))
+
+        when:
+        def change = multisetChangeAppender()
+                .calculateChanges(new RealNodePair(leftNode, rightNode), getProperty(SnapshotEntity, "multiSetValueObject"))
+
+        then:
+        assertThat(change)
+                .hasSize(1)
+                .hasValueRemoved(expectedVal)
+        where:
+        leftList <<  [[new DummyAddress(city: "New York"),
+                       new DummyAddress(city: "New York"),
+                       new DummyAddress(city: "Buffalo")] ]
+        rightList << [[new DummyAddress(city: "New York"),
+                       new DummyAddress(city: "New York")] ]
+        expectedVal << [valueObjectId(1, SnapshotEntity, "multiSetValueObject/4057abef011cfffbdbaa632f49dcef56")]
+    }
+
+    def "should throw exception on different multiset implementation"(){
+        when:
+        buildGraph(new SnapshotEntity(multiSetValueObject: HashMultiset.create([new DummyAddress(city: "New York")])))
+        buildGraph(new SnapshotEntity(multiSetValueObject: TreeMultiset.create([new DummyAddress(city: "New York")])))
+
+
+        then:
+        thrown(ClassCastException) //todo properly handle exception
```

Comment:
nie może lecieć wyjątek, jakie ma znaczenie czy to jest HashMultiset czy TreeMultiset? Powinieneś kodować do interfejsu Multiset a nie do implementacji